ssz/src/main/java/tech/pegasys/artemis/ssz/SSZTypes/Bitvector.java

```diff
@@ -13,94 +13,99 @@
 
 package tech.pegasys.artemis.ssz.SSZTypes;
 
-import static java.util.Objects.isNull;
+import static com.google.common.base.Preconditions.checkElementIndex;
 
-import java.util.Arrays;
+import com.google.common.base.Objects;
+import java.util.BitSet;
 import java.util.stream.IntStream;
 import org.apache.tuweni.bytes.Bytes;
 
 public class Bitvector {
 
-  private int size;
-  private byte[] byteArray;
+  private final BitSet data;
+  private final int size;
 
   public Bitvector(int size) {
-    this.byteArray = new byte[size];
+    this.data = new BitSet(size);
     this.size = size;
   }
 
-  public Bitvector(byte[] byteArray, int size) {
-    this.byteArray = byteArray;
+  public Bitvector(BitSet bitSet, int size) {
+    this.data = bitSet;
     this.size = size;
   }
 
+  public Bitvector(Bitvector bitvector) {
+    this.data = (BitSet) bitvector.data.clone();
+    this.size = bitvector.size;
+  }
+
   public void setBit(int i) {
```

Comment:
Would be perfect to make this class immutable to avoid potential modification side effects like this: 
https://github.com/PegaSysEng/teku/blob/c6e8e56b7a0edc078a88d1b40587262d22bd4c0a/ethereum/core/src/main/java/tech/pegasys/artemis/core/EpochProcessorUtil.java#L209-L221

Here a `Bitvector` copy is made just because of `.rightShift(1)`. Else it could have erroneous side effect .

That also relates to `Bitlist` but don't think this fits this PR

Reply:
Yeah makes total sense. I've made a JIRA ticket for this issue. https://pegasys1.atlassian.net/browse/BC-356

Reply:
I tried making `Bitlist` immutable when I converted it over but since the underlying `BitSet` is mutable and all of it's operation mutate the set it didn't really work out very well and I rolled it back.